File: server/src/game/conf/RandomName.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type RandomNameElem struct {
	Id int
	PlayerName string
	PlayerNameTw string
	PlayerNameEn string
	
}

type RandomName0 struct {
	Dic map[int]*RandomNameElem
	Arr []*RandomNameElem
}

var (
	RandomName RandomName0
)

func loadRandomName(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	RandomName.Dic = make(map[int]*RandomNameElem)
	RandomName.Arr = make([]*RandomNameElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &RandomName.Arr)
	for _,conf := range RandomName.Arr{
		RandomName.Dic[conf.Id] = conf
	}
	
}

File: tools/conf/output/ts/RandomName.ts
class RandomName {
    private static _dic:Dictionary<RandomName> = new Dictionary<RandomName>();
    private static _arr:Array<RandomName> = new Array<RandomName>();

	 public Id :number ;
     public PlayerName :string ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<RandomName>{
        return  RandomName._dic;
    }
    public static get Arr():Array<RandomName>{
        return RandomName._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:RandomName = <RandomName>json[i];
            RandomName._arr.push(config);
            RandomName._dic.add(config.Id,config);
        }
    }
}